### PR TITLE
Initial cut of create binding form

### PR DESCRIPTION
**Pull request by rhamilto**
TODO:
- [ ] add parameters

localhost_9000_k8s_ns_robb_serviceinstances_postgresql-persistent_create-binding 1

localhost_9000_k8s_ns_robb_serviceinstances_postgresql-persistent_create-binding

### CODE REVIEW

File: frontend/public/components/app.jsx [hunk 1]
```diff
@@ -137,6 +137,7 @@ class App extends React.PureComponent {
           <LazyRoute path={`/k8s/ns/:ns/${SubscriptionModel.plural}/new`} exact loader={() => import('./operator-lifecycle-manager').then(m => NamespaceFromURL(m.CreateSubscriptionYAML))} />
 
           <LazyRoute path="/k8s/cluster/clusterserviceclasses/:name/new" exact loader={() => import('./service-catalog/create-instance').then(m => m.CreateInstance)} />
+          <LazyRoute path="/k8s/ns/:ns/serviceinstances/:name/create-binding" exact loader={() => import('./service-catalog/create-binding').then(m => m.CreateBinding)} />
```
Comment (spadgett) [inline frontend/public/components/app.jsx]: 👍

File: frontend/public/components/service-catalog/create-binding.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  }
+
+  static getDerivedStateFromProps(props: CreateBindingFormProps, state: CreateBindingFormState) {
+    const { name } = state;
+    const { obj } = props;
+    const newState: any = {};
+
+    const defaultName = _.get(obj, 'data.metadata.name');
+    if (!name && defaultName) {
+      newState.name = defaultName;
```
Comment (spadgett) [inline frontend/public/components/service-catalog/create-binding.tsx]: I'd just use:

```jsx
return { name: defaultName };
```

and then `return null` below

File: frontend/public/components/service-catalog/create-binding.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        name: this.state.name,
+        namespace: _.get(this.props.obj, 'data.metadata.namespace'),
+      },
+      spec: {
+        instanceRef: {
+          name: _.get(this.props.obj, 'data.metadata.name'),
+        }
+      },
+    };
+    const ko = kindObj('ServiceBinding');
```
Comment (spadgett) [inline frontend/public/components/service-catalog/create-binding.tsx]: Pass the model instead of using `kindObj`

**Comment by spadgett** [comment]
We can make parameters a follow on

**Comment by spadgett** [comment]
I suggest showing some basic metadata about the service instance like name and namespace and a brief description of what bindings are at the top of the form

**Comment by rhamilto** [comment]
Thanks for the feedback, @spadgett.  Comments addressed.

File: frontend/public/components/utils/nav-title.tsx [hunk 4]
```diff
@@ -44,8 +44,10 @@ export const NavTitle = connectToModel((props: NavTitleProps) => {
     { breadcrumbsFor && !_.isEmpty(data) && <BreadCrumbs breadcrumbs={breadcrumbsFor(data)} /> }
     <h1 className={classNames('co-m-pane__heading', {'co-m-pane__heading--logo': isCSV})}>
       { logo }
-      { menuActions && !_.isEmpty(data) && !_.get(data.metadata, 'deletionTimestamp') && <ActionsMenu actions={menuActions.map(a => a(kindObj, data))} /> }
-      { buttonActions && !_.isEmpty(data) && <ActionButtons actionButtons={buttonActions.map(a => a(kindObj, data))} /> }
+      { (buttonActions || menuActions) && <div className="co-actions">
```
Comment (spadgett) [inline frontend/public/components/utils/nav-title.tsx]: I think this is a little cleaner and handles empty actions arrays.

```jsx
const hasButtonActions = !_.isEmpty(buttonActions);
const hasMenuActions = !_.isEmpty(menuActions);
const showActions = (hasButtonActions || hasMenuActions) && !_.isEmpty(data) && !_.get(data, 'deletionTimestamp');
// ...
{showActions && <div className="co-actions">
  { hasButtonActions && <ActionsButtons ... /> }
  { hasMenuActions && <ActionsMenu .../> }
</div>}
```

**Pull request merged**